app_server/app/controllers/StreamController.js

```diff
@@ -0,0 +1,143 @@
+
+function StreamController(server, options) {
+  this.server = server;
+  this.options = options;
+}
+
+var Class = StreamController.prototype;
+
+Class.registerRoutes = function () {
+  this.server.route({method: 'GET', path: '/',
+                     config: {
+                       validate: streamListParamsValidator,
+                       auth: {scope: Authenticator.SCOPE.ALL}
+                     },
+                     handler: this.getListOfStreams});
+
+  this.server.route({method: 'GET', path: '/{id}',
+                     config: {
+                       validate: singleStreamValidator,
+                       auth: {scope: Authenticator.SCOPE.ALL}
+                     },
+                     handler: this.getStreamById});
+
+  this.server.route({method: 'POST', path: '/',
+                     config: {
+                       validate: streamCreatePayloadValidator,
+                       auth: {scope: Authenticator.SCOPE.ALL}
+                     },
+                     handler: this.createStream});
+};
+
+/* Routes handlers */
+Class.createStream = function (request, reply) {
+  logger.debug('Creating stream');
+
+  var userId = request.auth.credentials.userId;
+  var currTime = Date.now();
+
+  return Promise.method(function generateStream() {
+    var newStream = {
+      title: request.payload.title,
+      description: request.payload.description,
+      createdAt: currTime,
+      appInstance: hash.update(userId + currTime).digest('hex')
+    };
+    return newStream;
+  })().then(function createNewStream(newStream) {
+    return Service.createNewStream(userId, newStream);
+  }).then(function(result) {
+    if (result instanceof CustomError.NotFoundError) {
+      logger.error('Stream could not be created');
+      reply(Boom.unauthorized(result.message));
+    }
+
+    reply(result);
+  }).catch(function(err) {
+    logger.error(err);
+    reply(Boom.unauthorized('User not found'));
+  });
+};
+
+Class.getStreamById = function (request, reply) {
+  logger.debug('Getting stream by Id');
+
+  Service.getStreamById(request.params.id).then(function(result) {
+    if (result instanceof Error) {
+      logger.error(result.message);
+      reply(Boom.notFound(result.message));
+      return;
+    }
+
+    reply(result);
+  });
+};
+
+Class.getListOfStreams = function (request, reply) {
+  logger.debug('Getting list of streams');
+
+  var filters = {
+    state: request.query.state ? request.query.state : 'live',
+    sort: request.query.sort ? request.query.sort : 'time',
+    order: request.query.order ? request.query.order : 'desc'
+  };
+
+  Service.getListOfStreams(filters).then(function(listStreams) {
+    if (!listStreams || listStreams instanceof Error) {
+      reply(Boom.unauthorized('Stream not found'));
+      return;
+    }
+
+    reply(listStreams);
+  });
+};
+
+/* Validator for routes */
+var singleStreamValidator = {
+  params: {
+    id: Joi.string().required()
+  }
```

Comment:
if stream's id is also a GUID then is this validator is preferably: `Joi.string().guid()`